pages/blog/posts/understanding-lexical-dynamic-scopes.md

```diff
@@ -0,0 +1,244 @@
+---
+title: "Understanding JSON Schema Lexical and Dynamic Scopes"
+date: "2024-01-19"
+tags:
+  - Fundamentals
+  - Scopes
+type: Engineering
+cover: /img/posts/2024/understanding-lexical-dynamic-scopes/cover.webp
+authors:
+  - name: Juan Cruz Viotti
+    photo: /img/avatars/jviotti.webp
+    link: https://www.jviotti.com
+    byline: O'Reilly author, founder of Intelligence.AI and Sourcemeta
+excerpt: "A deep dive on lexical and dynamic scopes, essential for understanding more advanced topics like dynamic referencing"
+---
+
+Most of the keywords defined by the JSON Schema organization can be either
+evaluated on their own, or by only considering the values of their adjacent
+keywords. For example, the `type` keyword is independent of any other keywords,
+while the `minContains` and `maxContains` keywords depend on the `contains`
+keyword defined in the same schema object.
+
+However, there is a small set of keywords whose evaluation depend on the
+<i>scope</i> they are in. These keywords are `$ref`, `$dynamicRef`,
```

Comment:
Any reason you're using tags instead of normal markdown syntax?

Reply:
They don't seem to work. `*` seems to remain as just literal `*` for some strange reason

Reply:
Hah, fun. OK, weird. I don't know anything about the markdown compilation toolchain here, so guess if this works..

Reply:
I typically use _ for italics, but I don't see a reason why * shouldn't work as well.  Odd.

Reply:
This should be a problem with the website component in charge on converting Markdown to HTML. I'll file an issue for this.

Reply:
This PR fixes this bug: https://github.com/json-schema-org/website/pull/275

Reply:
@gregsdennis I tried _ as well and it didn't work either 😅